### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Oklukeok**
### **_Features:_**
- Solved the N+1 problem when fetching all posts and their comments for a user.
- (Didn't use bullet gem).
- Created integration test with Capybara for:
  - User index page.
  - user show page.
  - User post index page.
  - Post show page.
### **_Logs:_**

**Before:**

image

**After:**

image

**Capybara testing:**

image

### CODE REVIEW

File: spec/integration/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'User index', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Lucca', photo: 'https://avatars.githubusercontent.com/u/108556234?v=4',
+                        bio: 'Teacher form Mexico.', posts_counter: 1)
+    visit users_path
+  end
```
Comment (codepantha) [inline spec/integration/users_index_spec.rb:8]: - [x] Please add a test that shows a redirect to the user's show page when you click on a user as per requirements 👍 
image

File: spec/integration/posts_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      @first_post = Post.create(author: @user, title: 'My post', text: 'This is my first post',
+                                comments_counter: 0, likes_counter: 0)
+      Comment.create(post: @first_post, author: @user2, text: 'This the first post comment')
+
+      visit user_posts_path(@user)
+    end
+
+    it 'should show the user profile picture' do
+      expect(page.body).to include(@user.photo)
+    end
```
Comment (codepantha) [inline spec/integration/posts_index_spec.rb:19]: - [x] Please add a test that shows a redirect to the post's show page when you click on a post as per requirements 👍
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Integration specs 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] All the integration tests seem to be failing. Please take a look at the screenshot below 👇 
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Oklukeok** [comment]
Hello Reviewer, I added the new tests and you can find in the PR description a new loom video showing the tests.
Thanks!

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Oklukeok, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 
- All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**